rust/connlib/snownet/src/node.rs

```diff
@@ -1434,6 +1434,8 @@ impl Connection {
                     source,
                     ..
                 } => {
+                    tracing::debug!(%destination, %source, "New nomination");
```

Comment:
You could add a more permanent log further down as an alternative to if the socket is the same. That would avoid a duplicate log in case we are actually updating the socket.

Reply:
I wanted to see if we aborted with something before that, I think it can only happen with the `relayed` branch but seems useful to debug #4358

Reply:
Ah yes, that probably deserves a `warn` log so we also come across it in production?